modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/RoleResourceV2Manager.java

```diff
@@ -328,6 +325,179 @@ public SCIMResponse updateWithPATCHRole(String id, String patchRequest, RoleV2Ma
         }
     }
 
+    @Override
+    public SCIMResponse createRoleMeta(String postRequest, RoleV2Manager roleManager) {
+
+        try {
+            validateManager(roleManager);
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getRoleResourceV3Schema();
+            RoleV2 role = decodeRole(postRequest, schema);
+            ServerSideValidator.validateCreatedSCIMObject(role, SCIMSchemaDefinitions.SCIM_ROLE_V2_SCHEMA);
+
+            RoleV2 createdRole = roleManager.createRoleMeta(role);
+            return buildSCIMResponse(createdRole, SCIMConstants.ROLE_V3_ENDPOINT);
+        } catch (InternalErrorException | BadRequestException | ConflictException | CharonException | NotFoundException
+                 | NotImplementedException e) {
+            return encodeSCIMException(e);
+        }
+    }
+
+    @Override
+    public SCIMResponse updateWithPUTRoleMeta(String id, String putRequest, RoleV2Manager roleManager) {
+
+        try {
+            validateManager(roleManager);
+            JSONEncoder encoder = getEncoder();
+            SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getRoleResourceV3Schema();
```

Comment:
why we have SCIM_ROLE_V2_SCHEMA in the previous method and using getRoleResourceV3Schema here?

Reply:
Fixed with : https://github.com/wso2/charon/pull/435/commits/adb417717e375a2b27daa8200ee3b635f49e8332